src/btree/bt_sync_obsolete.c

```diff
@@ -392,29 +426,40 @@ __checkpoint_cleanup_page_skip(
     }
 
     /*
-     * The checkpoint cleanup fast deletes the obsolete leaf page by marking it as deleted
-     * in the internal page. To achieve this,
+     * From this point, the page is on disk and checkpoint cleanup should NOT induce a read if at
+     * least of one of the following conditions is met:
      *
-     * 1. Checkpoint has to read all the internal pages that have obsolete leaf pages.
-     *    To limit the reading of number of internal pages, the aggregated stop durable timestamp
-     *    is checked except when the table is logged. Logged tables do not use timestamps.
+     * - The page is a leaf page with no overflow items because obsolete leaf pages with overflow
+     * keys/values cannot be fast deleted to free the overflow blocks. The overflow blocks can be
+     * freed during reconciliation after being marked dirty.
      *
-     * 2. Obsolete leaf pages with overflow keys/values cannot be fast deleted to free
-     *    the overflow blocks. Read the page into cache and mark it dirty to remove the
-     *    overflow blocks during reconciliation.
+     * - The page is not fully deleted (checked using the aggregated stop durable timestamp) and
+     *   - the table is not logged (logged tables do not use timestamps) or checkpoint cleanup is
+     * configured to reclaim space.
      *
      * FIXME: Read internal pages from non-logged tables when the remove/truncate
      * operation is performed using no timestamp.
      */
+    aggressive_mode = F_ISSET(S2C(session), WT_CONN_CKPT_CLEANUP_SKIP_INT);
+    logged = F_ISSET(S2BT(session), WT_BTREE_LOGGED);
+    page_fully_deleted = addr.ta.newest_stop_durable_ts != WT_TS_NONE;
```

Comment:
This doesn't tell whether the page is fully deleted or not. Change the variable name to `has_deletes` or something.

Reply:
Thank you, see 813f7b864219d2c8b759b1320c5944025e83fd36.